agent/engine/dependencygraph/graph.go

```diff
@@ -193,6 +182,34 @@ func verifyStatusResolvable(target *apicontainer.Container, existingContainers m
 	return true
 }
 
+// verifyContainerOrderingStatusResolvable validates that `target` can be resolved given that
+// the dependsOn containers are resolved and there are `existingContainers`
+// (map from name to container). The `resolves` function passed should return true if the named container is resolved.
+
+func verifyContainerOrderingStatusResolvable(target *apicontainer.Container, existingContainers map[string]*apicontainer.Container,
+	resolves func(*apicontainer.Container, *apicontainer.Container, string) bool) bool {
+
+	targetGoal := target.GetDesiredStatus()
+	if targetGoal != target.GetSteadyStateStatus() && targetGoal != apicontainerstatus.ContainerCreated {
+		// A container can always stop, die, or reach whatever other state it
+		// wants regardless of what dependencies it has
+		return true
+	}
+
+	for _, dependency := range target.DependsOn {
+		maybeResolves, exists := existingContainers[dependency.Container]
+		if !exists {
+			return false
+		}
+		mayBeResolvedStatus := dependency.Condition
+		if !resolves(target, maybeResolves, mayBeResolvedStatus) {
+			log.Warnf("Failed to resolve the dependency [%v] for the target [%v]", maybeResolves, target)
```

Comment:
Actually, I'd get rid of this log message and just return an error built from this string.